include/flatbuffers/idl.h

```diff
@@ -268,11 +268,13 @@ struct Definition {
         index(-1),
         refcount(1) {}
 
+  virtual ~Definition() {}
+
   flatbuffers::Offset<
       flatbuffers::Vector<flatbuffers::Offset<reflection::KeyValue>>>
-  SerializeAttributes(FlatBufferBuilder *builder, const Parser &parser) const;
+  virtual SerializeAttributes(FlatBufferBuilder *builder, const Parser &parser) const;
```

Comment:
Why these methods are `virtual`?
I don't see any overloads of these methods.

Reply:
It was needed to allow dynamic cast which is used to determine the subclass type at runtime here:
https://github.com/bjornharrtell/flatbuffers/blob/676b2135bfaa1853dfbb06c92b5c16a0d81bb13a/src/idl_gen_ts.cpp#L507-L515

Reply:
On every call of `AddImport` you downcast a well-known type to the `Definition` and then use dynamic_cast to check the orinal type. On to this was added a virtual table to the pure non-virtual struct.
It looks strange a little bit.
Maybe add a `MakeImportStatement` method and use its result?

Reply:
I agree it's a bit ugly and it would be nicer to have the two methods:

std::string AddImport(import_set &imports, const Definition &dependent, const StructDef &dependency)
std::string AddImport(import_set &imports, const Definition &dependent, const EnumDef &dependency)

And in that way avoid the casting issue. But I find it hard to rewrite/breakout the logic of AddImport to avoid duplicating the code.

Reply:
But I've went with the above anyways with some code duplication and removed the changes in idl.h.

Reply:
Possible solutions:
```cpp
std::string MakeImportStatment(const StructDef &dependency);
std::string MakeImportStatment(const EnumDef &dependency);
std::string AddImportImpl(import_set &imports, const Definition &dependent, const Definition &dependency, std::string import_statment);

std::string AddImport(import_set &imports, const Definition &dependent, const StructDef &dependency) {
  return AddImportImpl();
}
std::string AddImport(import_set &imports, const Definition &dependent, const EnumDef &dependency) {
  return AddImportImpl();
}
```
This solution also a bit ugly but it is local and doesn't disturb `idl.h`.